raft-tech/TANF-app
# Fix/1641 stt code in user

## Summary of Changes
Pull request closes #1641
* update the django admin user details page to make the relationship between a user and the STT they are associated with more obvious 
   * dropdown select list for stt (if stt user, deactivated if regional user)
   * [STT Name](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/tdrs-backend/tdpservice/stts/management/commands/data/states.csv) field (e.g. Virginia) appears in user profile
   * [STT_code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/tdrs-backend/tdpservice/stts/management/commands/data/states.csv) field (e.g. 11) appears as location ID in user profile

This change removes the `location_type` and `location_id` fields managed by the `location` `GenericForeignKey` in favor of separate `region` and `stt` fields on the `user` model. This reduces complexity some - the major trickiness is around validation - and allows us to simplify the admin experience of selecting an stt location. The django admin User page was updated to allow an `stt` to both display and be searched by `name` or `stt_code`.

## How to Test
```
cd tdrs-frontend && npm run start
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
2. Request access via the form, select a location.
3. In the Django Admin, edit the user and verify that the selected location matches what was entered in the Request Access form.
4. Select the Data Analyst role, select the "Pending" approval status, and save the user. Verify on the frontend that the "Profile" tab becomes visible and the "Data Files" tab is still not accessible. The "Profile" tab should show the name of the user's assigned location.
5. In the Django Admin, select the "Approved" approval status for the user, verify the "Data Files" tab becomes visible on the frontend.
6. Navigate to the Data Files tab, submit a data file and verify that the STT associated with the data file matches the user's STT.
7. In the Django Admin, select a new location for the user.
8. In the frontend, navigate to the Data Files tab and submit a new data file. Verify the associated STT matches the user's new location.
9. In the Django Admin, select the "OFA Regional Staff" group and remove the user's STT assignment. Save the user (do not add a Region yet\*).
10. Now add a region, and save the user. Verify that the user's new location appears in the "Profile" tab on the frontend.
11. Navigate to the "Data Files" tab. Verify a message is shown stating that no STT is assigned to the user.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] update the django admin user details page to make the relationship between a user and the STT they are associated with more obvious 
   * dropdown select list for stt (if stt user, deactivated if regional user)
   * [STT Name](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/tdrs-backend/tdpservice/stts/management/commands/data/states.csv) field (e.g. Virginia) appears in user profile
   * [STT_code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/tdrs-backend/tdpservice/stts/management/commands/data/states.csv) field (e.g. 11) appears as location ID in user profile

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
